### PR TITLE
[AppBar] Annotate MDCAppBarNavigationControllerDelegate's appBarNavigationController:willAddAppBar:asChildOfViewController: as to be deprecated.

### PR DESCRIPTION
**Pull request by jverkoey**
Part of https://github.com/material-components/material-components-ios/issues/9392

### CODE REVIEW

File: components/AppBar/src/MDCAppBarNavigationController.h [hunk 1]
```diff
@@ -19,10 +19,42 @@
 @class MDCAppBarNavigationController;
 @class MDCFlexibleHeaderViewController;
 
+/**
+ Defines delegate methods that will be deprecated.
+ */
+@protocol MDCAppBarNavigationControllerToBeDeprecatedDelegate <NSObject>
```
Comment (romoore) [inline components/AppBar/src/MDCAppBarNavigationController.h:25]: Why do you need to put these in a separate protocol? Isn't it sufficient to just put a comment marking them as "To Be Deprecated"?

Comment (jverkoey) [reply components/AppBar/src/MDCAppBarNavigationController.h:25]: This approach will enable us to detect that the symbol is to be deprecated through symbol analysis. Comments alone unfortunately are not picked up by symbol analysis. That being said, I will take a look to see if Kythe does support some level of comment analysis before merging this PR.

Comment (jverkoey) [reply components/AppBar/src/MDCAppBarNavigationController.h:25]: I've researched the feasibility of extracting docs and have concluded that we won't be able to make use of them for our reporting purposes.

As such, this PR is in line with our current approach to annotating and tracking to-be-deprecated APIs, insofar as requiring that the words "ToBeDeprecated" appear somewhere in the symbol definition (either as the protocol name or as a class category).

Comment (randallli) [reply components/AppBar/src/MDCAppBarNavigationController.h:25]: Where can I read about `adding "ToBeDeprecated" somewhere in the symbol definition` to catchup on why we are doing this?

Comment (randallli) [reply components/AppBar/src/MDCAppBarNavigationController.h:25]: Found it:
https://github.com/material-components/material-components-ios/blob/cb6b8e9599e9018c545e0011a3ded82597988106/contributing/deprecation_policy.md#step-4-add-a-comment-to-the-api-indicating-that-it-will-be-deprecated

This is the closest thing I found to a category protocol. doesn't work.
https://stackoverflow.com/questions/1521267/defining-categories-for-protocols-in-objective-c

Comment (jverkoey) [reply components/AppBar/src/MDCAppBarNavigationController.h:25]: Yep; kinda bummed I'm not aware of another scalable way to annotate protocol APIs as to-be-deprecated; this is the best I could come up with.

**Review by randallli — APPROVED** [review]

**Pull request merged**